File: BaekjoonOnlineJudge/2501_약수구하기.cpp
#include <cstdio>
int ans[10000];
int main() {
	int n, k, s = 0, i = 0;
	scanf("%d %d", &n, &k);
	while (s < n / 2) {
		if (n%++s == 0)
			ans[i++] = s;
	}
	ans[i] = n;
	printf("%d", ans[k - 1]);
	return 0;
}

File: BaekjoonOnlineJudge/2501_약수구하기.java
class Main{
    public static void main(String[] args) {
        int[] ans=new int[10000];
    	int t=0, i=0;
    	java.util.Scanner s=new java.util.Scanner(System.in);
        int n=s.nextInt();
        int k=s.nextInt();
        s.close();
    	while(t<n/2) {
    		if (n%++t == 0)
    			ans[i++] = t;
    	}
    	ans[i] = n;
    	System.out.println(ans[k-1]);
    }
}